packages/schematics/src/command-line/affected-apps.ts

```diff
@@ -31,27 +36,43 @@ export type DepGraph = {
 };
 
 export function touchedProjects(
+  implicitDependencies: ImplicitDependencies,
   projects: ProjectNode[],
   touchedFiles: string[]
 ) {
   projects = normalizeProjects(projects);
   touchedFiles = normalizeFiles(touchedFiles);
-  return touchedFiles.map(f => {
-    const p = projects.filter(project => project.files.indexOf(f) > -1)[0];
-    return p ? p.name : null;
-  });
+  return Array.from(
+    touchedFiles
+      .map(f => {
+        const p = projects.find(project => {
+          return project.files.indexOf(f) > -1;
+        });
+        if (p) {
+          return [p.name];
+        }
+        return (implicitDependencies[f] as string[]) || [];
+      })
+      .reduce((touchedProjects, items) => {
+        items.forEach(item => {
+          touchedProjects.add(item);
+        });
+        return touchedProjects;
+      }, new Set<string>())
+  );
 }
 
 function affectedProjects(
   npmScope: string,
   projects: ProjectNode[],
+  implicitDependencies: ImplicitDependencies,
   fileRead: (s: string) => string,
   touchedFiles: string[]
 ): ProjectNode[] {
   projects = normalizeProjects(projects);
   const deps = dependencies(npmScope, projects, fileRead);
-  const tp = touchedProjects(projects, touchedFiles);
-  if (tp.indexOf(null) > -1) {
+  const tp = touchedProjects(implicitDependencies, projects, touchedFiles);
+  if (tp.indexOf('*') > -1) {
```

Comment:
Why would it return '*'? I thought you are doing `nxJson.implicitDependencies[key] = projects.map(p => p.name)`, so wildcards should go away right after you read the file.